pkg/kfapp/gcp/gcp.go

```diff
@@ -727,62 +729,65 @@ func (gcp *Gcp) updateDM(resources kftypesv3.ResourceEnum) error {
 	if err = blockingWait(gcp.kfDef.Spec.Project, deploymentmanagerService, dmOperationEntries); err != nil {
 		return kfapis.NewKfErrorWithMessage(err, "could not update deployment manager entries")
 	}
-	exp := backoff.NewExponentialBackOff()
-	exp.InitialInterval = 1 * time.Second
-	exp.MaxInterval = 3 * time.Second
-	exp.MaxElapsedTime = time.Minute
-	exp.Reset()
-	err = backoff.Retry(func() error {
-		// Get current policy
-		policy, policyErr := utils.GetIamPolicy(gcp.kfDef.Spec.Project, gcpClient)
-		if policyErr != nil {
-			return kfapis.NewKfErrorWithMessage(policyErr, "GetIamPolicy error")
-		}
-		utils.ClearIamPolicy(policy, gcp.kfDef.Name, gcp.kfDef.Spec.Project)
-		if err := utils.SetIamPolicy(gcp.kfDef.Spec.Project, policy, gcpClient); err != nil {
-			return kfapis.NewKfErrorWithMessage(err, "Set Cleared IamPolicy error: %v")
-		}
-		return nil
-	}, exp)
-	if err != nil {
-		return err
-	}
 
-	appDir := gcp.kfDef.Spec.AppDir
-	gcpConfigDir := path.Join(appDir, GCP_CONFIG)
-	iamPolicy, iamPolicyErr := utils.ReadIamBindingsYAML(
-		filepath.Join(gcpConfigDir, "iam_bindings.yaml"))
-	if iamPolicyErr != nil {
-		return &kfapis.KfError{
-			Code: iamPolicyErr.(*kfapis.KfError).Code,
-			Message: fmt.Sprintf("Read IAM policy YAML error: %v",
-				iamPolicyErr.(*kfapis.KfError).Message),
+	if _, iamStatErr := os.Stat(path.Join(gcp.kfDef.Spec.AppDir, GCP_CONFIG, "iam_bindings.yaml")); iamStatErr == nil {
```

Comment:
Please add a comment explaining why we check if the file exists.

Should you check that iamStatErr is an error other than file doesn't exist and if it is return that error?